src/solvers/base_solver.hpp

```diff
@@ -7,6 +7,8 @@
 #ifndef BASE_SOLVER
 #define BASE_SOLVER
 
+#include <map>
```

Comment:
Is this still needed?

Reply:
good call, I missed this one when reverting the std::map stuff